### PR TITLE
MM-52639 : Add load test for removed refetching of channels & channelMembers on browser focus

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-52639

### CODE REVIEW

**Comment by agnivade** [comment]
This looks good. Can we open this for review?

**Review by agnivade — APPROVED** [review]

**Comment by M-ZubairAhmed** [comment]
> Should we just wrap the whole thing under the if? In particular, did your server PR remove the call to view the channel as well (line 1076 here)?

No the call to viewChannel is not removed, its what `markChannelAsReadOnFocus` calls internally
https://github.com/mattermost/mattermost-server/blob/e300cb942adaea3c32d5e9f9ed71a7eae93392c2/webapp/channels/src/components/team_controller/team_controller.tsx#L90-L93

**Comment by agnivade** [comment]
@M-ZubairAhmed - Were you able to run a full load test with this?

File: loadtest/control/simulcontroller/actions.go [hunk 1]
```diff
@@ -1057,12 +1057,15 @@ func unreadCheck(u user.User) control.UserActionResponse {
...
+		}
 
-	if err := u.GetChannelMembersForUser(u.Store().Id(), team.Id); err != nil {
-		return control.UserActionResponse{Err: control.NewUserError(err)}
+		if err := u.GetChannelMembersForUser(u.Store().Id(), team.Id); err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
 	}
 
 	collapsedThreads, resp := control.CollapsedThreadsEnabled(u)
```
[the remark this answers is not in the document]
Comment (agarciamontoro) [reply loadtest/control/simulcontroller/actions.go:1071]: @M-ZubairAhmed, sorry for the delay here. What do you mean with "The complete viewChannel actions are only called when `channel != nil`"? 
To make sure we're on the same page, let me share my understanding: what this action models is the set of requests done when the user has the window unfocused for some time and then focus it again. If I'm not mistaken, this set of requests are triggered by [`handleFocus` in the team controller](https://github.com/mattermost/mattermost/blob/9a55280d7acf3013f06114727b64278fbe6e9d2d/webapp/channels/src/components/team_controller/team_controller.tsx#L86-L106), which is what your PR wrapped with a check on the new  `disableRefetchingOnBrowserFocus`. That's why I proposed wrapping the whole action here with the same check.
Is my mental model different from what you had in mind?

[the remark this answers is not in the document]
Comment (M-ZubairAhmed) [reply loadtest/control/simulcontroller/actions.go:1025]: @agarciamontoro i agree with Agniva,
Yes you are correct that flag `disableRefetchingOnBrowserFocus` only disables the refetching of `GetChannelsForTeamForUser` & `GetChannelMembersForUser` as done in this PR.

Coming to another point of 
> the complete viewChannel actions are only called when channel != nil"

I am sorry I got confused with another action, please ignore

**Comment by agarciamontoro** [comment]
@M-ZubairAhmed, would you be open to run a load-test with and without the feature flag to check the differences?

**Comment by agarciamontoro** [comment]
Hey, @M-ZubairAhmed, let us know if we can help move this forward! Thank you :pray:

**Comment by agnivade** [comment]
@M-ZubairAhmed - How is it going with this?

**Comment by M-ZubairAhmed** [comment]
Its not in my radar currently, feel free to close it

**Pull request closed without merging**